pkg/sessionctx/variable/sysvar.go

```diff
@@ -2087,7 +2087,7 @@ var defaultSysVars = []*SysVar{
 	}},
 	{Scope: ScopeGlobal | ScopeSession, Name: TiDBRedactLog, Value: BoolToOnOff(DefTiDBRedactLog), Type: TypeBool, SetSession: func(s *SessionVars, val string) error {
 		s.EnableRedactLog = TiDBOptOn(val)
-		errors.RedactLogEnabled.Store(s.EnableRedactLog)
+		errors.RedactLogEnabled.Store(val)
```

Comment:
current master don't call `TiDBOptOn`, which doesn't support `MARKER`, see https://github.com/pingcap/tidb/blob/9657f0631686a346e2a7f5d22d6e640df7499bb3/pkg/sessionctx/variable/sysvar.go#L2283-L2284

Reply:
* https://github.com/pingcap/tidb/pull/51642

This PR is not cherry-pick back to the release-7.5 branch, so still use `TIDBOptOn`